### ISSUE TITLE
OH3: Item with type number:temperatur wrongly formatted in default widget item

### ISSUE THREAD

**Issue by FSeidinger** [openhab/openhab-core#2039]
I'm using openHAB 3.0 with docker on a linux host OS.

I've linked a `number:temperature` item via KNX bridge to a heating actor object with DPT 9.002 to shift the nominal value. I've noticed to issues with this configuration

1. The value of the item is a floating point number in degrees celsius. When sending the command `1.5` to the item, the internal state is `1.5 °C`, but the default item widget, e.g. when selecting the item in the semantic model, formats this value as `2`. So the unit is missing and it is only shown as integer (or rounded?). See attached image for details.
1. When sending a command to the item via API, the item gets updated, but no telegram is send to the KNX bus.

![image](https://user-images.githubusercontent.com/4890418/103422212-931e0200-4ba0-11eb-8498-cfb0bb1629c9.png)

As you can see in the next image, the formatting in the item list is correct as floating point and unit.

![image](https://user-images.githubusercontent.com/4890418/103422319-248d7400-4ba1-11eb-82c9-b3fda490cdf5.png)

BTW, how can the unit be changed to Kelvin (K) which is the right unit for DPT 9.002 according the the KNX specification.

**Comment by brangc** [comment]
Unsubscribe

Frank Seidinger <notifications@github.com> schrieb am Do. 31. Dez. 2020 um
19:49:

**Comment by cweitkamp** [comment]
@ghys Do you think this is an UI related issue or similar tohttps://github.com/openhab/openhab-core/issues/2037#issuecomment-753370797?

**Comment by ghys** [comment]
I tjink I recall it was the KNX binding that added a %d pattern to the state description so the displayed state loses the decimals and the unit.

**Comment by FSeidinger** [comment]
> I think I recall it was the KNX binding that added a %d pattern to the state description so the displayed state loses the decimals and the unit.

That sound about right. I will investigate that a little further and if this is a knx binding problem with OH3, I will forward this issue to the addons project.

**Comment by FSeidinger** [comment]
> @ghys Do you think this is an UI related issue or similar to [#2037 (comment)](https://github.com/openhab/openhab-core/issues/2037#issuecomment-753370797)?

Nah, I'm quite sure that the issue #2037 is related to the profile assigned, whereas this issue is simply giving a wrongly formatted value.

**Comment by ghys** [comment]
In the meantime I think you can possibly override the channel-provided pattern by adding stateDescription metadata to the item - even maybe convert to Kelvins with something like `%. 1f °K`

**Comment by FSeidinger** [comment]
After diving in the KNX binding I now believe that the described problem is not to the KNX binding alone and can be at least be described an inconvenience.

There are three parts that come together. The Thing, the linked channel and the item. Currently bindings have, I guess, no way of telling the OH platform of formatting rules and valid ranges of thing states. Therefore if creating an item and linking it to a thing, there is not suitable way of setting the formatting rule.

In the past I often configured things and items via text files and because the channel link had no concept there, I nearly always have put formatting rules into the label of the item.

When doing the same via UI, you create the thing, and then either create the item and link it or you use the convenience functions like "Create Points from Thing". Putting the formatting in the label doesn't work that way anymore. Therefore one or two other steps are necessary:

1. Add a state description in the meta date like `%.1f °C` or `%.1f %unit%`
2. If using the `%unit%` approach you have to change the item type to `Number:Temperature`

IMHO the `Create Points from Thing` dialog should give you the hints and possibility to do both in an easy way. For this reason I would like to see this use case as an enhancement to usability (UX).

**Comment by Rossko57** [comment]
> Currently bindings have, I guess, no way of telling the OH platform of formatting rules and valid ranges of thing states.

hmm, bindings _can_ suggest `state descriptions` for each channel.   That's about channels, not Things, which do not have states.   The binding would need to 'know' some value represented a temperature in certain units, but I guess for KNX that is implied in (some) DPTs?  Does the DPT also carry information about how many decimal places to display, or is that really a decision for the user?

> I've linked a `number:temperature` item via KNX bridge

So your original problem is about linking a Number:Temperature type Item to a plain number type channel, which will not work well.

> 2\. When sending a command to the item via API

Likely the same problem, KNX binding plain number channel not accepting Quantity Type commands with units.

Basically, I think you're asking for an enhancement to KNX binding to add quantity type channels for some DPT?
